mlir/lib/IR/AsmPrinter.cpp

```diff
@@ -545,6 +545,10 @@ class SymbolAlias {
   bool isType : 1;
   /// A flag indicating whether this alias may be deferred or not.
   bool isDeferrable : 1;
+
+public:
+  /// Used to avoid printing incomplete aliases for recursive types.
+  bool isPrinted = false;
```

Comment:
On 64-bit systems, this won't increase the size of `SymbolAlias`, since there was already padding to align to 8 bytes (because of the StringRef). I'm happy to reduce the size of suffixIndex by 1 bit (29 bits should still be more than enough for it, hopefully) and pack this field into the prior bitfield if it's preferred though.

Reply:
Additionally, the bitfield won't be packed on Windows anyway because the MSVC ABI only packs bitfields when all members have the same type, but I'm not sure how much MLIR cares about that.

Reply:
Textual parsing/printing isn't considered to be performance-sensitive components.